### PR TITLE
Bug fixes in image converter

### PR DESCRIPTION
**Pull request by robwin311**
Hi,

I fixed some bugs in image converter. Unfortunately I haven't got enough time to resolve all problems and warnings, however I've marked all warnings that I detected in code.

### Fixes:
- The converter didn't work when running offline, because there were no $cf variable declared. I've added that.
- There were some wrong calls to class constants. Inside class they should be preceded by `self::`, outside class by `class_variable::`
- Some code remains from previous converter, some code was unnecessary.
- Because of no `$this->` inside dithering section, it do not work. Now it should!

All warnings was suppressed in order to detect all of them. They are marked with `// TODO`

### CODE REVIEW

**Comment by robwin311** [comment]
Uhh... I see one mistake. Give me a second.

File: img_conv_core.php [hunk 1]
```diff
@@ -73,13 +71,13 @@ function __construct ($path, $real_name, $out_name ,$dith) {
...
-        
+
         $this->r_earr = array();  /*Classification error for next row of pixels*/
         $this->g_earr = array();
         $this->b_earr = array();
-        
+
         if($this->dith) {
-            for($i = 0; $i < $w + 2; ++$i){
+            for($i = 0; $i < $this->w + 2; ++$i){
```
Comment (robwin311) [inline img_conv_core.php:80]: Because of that small mistake dithering did not work.

File: img_conv_core.php [hunk 2]
```diff
@@ -88,38 +86,39 @@ function __construct ($path, $real_name, $out_name ,$dith) {
...
             $img_tmp = imagecreatetruecolor($this->w, $this->h);
             imagecopy ($img_tmp, $this->img, 0 , 0 , 0 , 0, $this->w , $this->h);
             imagetruecolortopalette($this->img, false, $palette_size);
-           
+
 	        for($i = 0; $i < $palette_size; $i++) {
-	           $c = imagecolorsforindex ($this->img , $i);
-               array_push($this->d_out, $c['blue'], $c['green'], $c['red'], 0xFF); 
-	        }         
+               // TODO !WARNING SUPPRESSED! Warning when: format=indexed_8
```
Comment (robwin311) [inline img_conv_core.php:116]: I marked warnings this way.

File: img_conv_core.php [hunk 3]
```diff
@@ -651,140 +656,139 @@ private function classify_pixel($value, $bits){
...
-    $transp = "none";
-  }
-
   if(isset($_POST["dith"])){
     $dith = $_POST["dith"];
   }
   else {
     $dith = "enabled";
   }
+
```
Comment (robwin311) [inline img_conv_core.php]: Because of that it was impossible to run converter offline.

File: img_conv_core.php [hunk 4]
```diff
@@ -128,34 +127,35 @@ function convert($cf, $alpha = 0) {
                 $this->conv_px($x, $y);
             }
         }
-        
-        imagecopy ($this->img, $img_tmp,  0 , 0 , 0 , 0 , $this->w , $this->h);
+
+        // TODO !WARNING SUPPRESSED!
```
Comment (robwin311) [inline img_conv_core.php]: This warning worries me the most. In some cases Null is copied into $this->img. Maybe it should be surrounded with if statement?

File: img_conv_core.php [hunk 5]
```diff
@@ -128,34 +127,35 @@ function convert($cf, $alpha = 0) {
...
+        @imagecopy ($this->img, $img_tmp,  0 , 0 , 0 , 0 , $this->w , $this->h);
```
Comment (robwin311) [inline img_conv_core.php:132]: This warning worries me the most. In some cases Null is copied into $this->img. Maybe it should be surrounded with if statement?

**Comment by kisvegabor** [comment]
Hi Robert,

Thank you very much for the fixes! 
I will check the script offline to see the warning messages.

**Comment by kisvegabor** [comment]
I fixed the mentioned warnings. Can you review it? 
https://github.com/littlevgl/lv_utils/commit/97da65a9e1b7eb08379a4739c5b0495c6f2b779f

**Pull request merged**